### PR TITLE
MOTOR-699 Update PyMongo dependency to 3.12+

### PR DESCRIPTION
**Pull request by prashantmital**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Can you update the Compatibility Matrix too?

**Comment by prashantmital** [comment]
Done.

File: doc/requirements.rst [hunk 1]
```diff
@@ -45,6 +45,8 @@ Motor and PyMongo
 +-------------------+-----------------+
 | 2.4               | 3.11+           |
 +-------------------+-----------------+
+| 2.5               | 3.12+           |
++-------------------+-----------------+
 
 Motor and MongoDB
```
Comment (ShaneHarvey) [inline doc/requirements.rst:51]: Can you update this one too?

Comment (prashantmital) [reply doc/requirements.rst:51]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**